### PR TITLE
add warning for inaccurate MeanRadiusValue #4111

### PR DESCRIPTION
**Pull request by tgiroux**
## Description

Adds a simple check comparing the user-entered MeanRadiusValue and the localRadius of some control point to see if they are within a reasonable range. If not, a warning is output to the user.

## Related Issue

#4111

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jlaura** [comment]
@tgiroux What is the status on this? Does this need to be a draft still?

**Comment by tgiroux** [comment]
> @tgiroux What is the status on this? Does this need to be a draft still?

It's still not working. I remember not knowing quite how to implement this warning and everything I tried seemed to blow up. IIRC this was definitely something we wanted done for support but I was having a lot of trouble with it.

**Comment by AustinSanders** [comment]
@tgiroux Is this something that you'd want to look into together?  I'm on continuous support, so I've got hours allocated specifically for stuff like this.

File: isis/src/control/objs/BundleAdjust/BundleAdjust.cpp [hunk 1]
```diff
@@ -495,6 +495,30 @@ namespace Isis {
...
+
+        // Ensure aprioriRadius is within some threshold
+        // Considered potentially inaccurate if it's off by atleast a factor of two
+        if (aprioriRadius >= 2 * localRadius || aprioriRadius <= localRadius / 2) {
+            isMeanRadiusValid = false;
+        }
+
+        // Warn user for abnormal MeanRadiusValue
+        if (!isMeanRadiusValid) {
+          outputBundleStatus("Warning: User-entered MeanRadiusValue may be inaccurate\n");
```
Comment (acpaquette) [inline isis/src/control/objs/BundleAdjust/BundleAdjust.cpp]: I think we may want a slightly stronger warning here. Something like, "This option may introduce inaccuracy into the overall bundle adjustment." Still a solid change overall!

**Review by acpaquette — COMMENTED** [review]
Small suggestion to talk about but other than that this looks solid!

File: isis/src/control/objs/BundleAdjust/BundleAdjust.cpp (hunk 1 above)
Comment (tgiroux) [reply isis/src/control/objs/BundleAdjust/BundleAdjust.cpp]: I added a warning that outputs for any target body radii solve, and added some more detail on the warning for inaccurate mean radius input.

**Comment by KrisBecker** [comment]
Hi @tgiroux, just wanted to point out that this may cause issues with highly irregular bodies such as Eros where the major axis is more than twice the minors. I realize this is just a warning, but you may see a ton of these messages when running jigsaw if you solve for body parameters.

Here is the radii for Eros:

`
BODY2000433_RADII                  = (17.0, 5.5, 5.5)
`

**Comment by tgiroux** [comment]
> Hi @tgiroux, just wanted to point out that this may cause issues with highly irregular bodies such as Eros where the major axis is more than twice the minors. I realize this is just a warning, but you may see a ton of these messages when running jigsaw if you solve for body parameters.
> 
> Here is the radii for Eros:
> 
> `BODY2000433_RADII = (17.0, 5.5, 5.5)`

Thanks for pointing that out. I think it should be OK for this warning to occur for irregular bodies since it appears just once at the beginning of the output and only states that the value appears to be inaccurate. If the warning raises concern it could quickly be accounted for by checking the target body parameters for a mistake/typo.

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
Solving for mean radius on such an elliptical body is almost meaningless. I do not think having some extra warnings in that case is a problem.

**Pull request merged**